code/game/objects/structures/morgue.dm

```diff
@@ -97,6 +106,159 @@
 	toggle_morgue(user)
 
 
+//Lifted from Unity stasis.dm and refactored. ~Zuhayr
+/obj/structure/morgue/process()
```

Comment:
@Rohesie I'll be honest, I submitted this PR as one file because I didn't want to use the git to submit multiple files.

Also, I intentionally made it so that dead people getting cryoed only retrieves stuff and doesn't unlock their role: If a marine perma dies, that slot is supposed to be gone; unless of course we want to change the design philosophy and encourage body retrieval which I'm not at all opposed to.

Reply:
That is good. Body retrieval should warrant a little more discussion, I guess, but it's not in the scope of this PR.

What I meant is in the guideline as "No duplicated code". This will require modifying more than one file, unfortunately.